zipline/pipeline/loaders/blaze/core.py

```diff
@@ -729,25 +729,36 @@ def __new__(cls,
             expr,
             deltas,
             checkpoints,
-            odo_kwargs or {},
+            frozenset((odo_kwargs or {}).items()),
         )
 
     def __repr__(self):
         # If the expressions have _resources() then the repr will
         # drive computation so we take the str here.
-        cls = type(self)
-        return super(ExprData, cls).__repr__(cls(
+        return repr(_expr_data_base(
             str(self.expr),
             str(self.deltas),
             str(self.checkpoints),
-            self.odo_kwargs,
+            dict(self.odo_kwargs),
         ))
 
-    def __hash__(self):
-        return id(self)
+    @staticmethod
+    def _expr_eq(a, b):
+        return a is b is None or a.isidentical(b)
 
     def __eq__(self, other):
-        return self is other
+        if not isinstance(other, ExprData):
+            return NotImplemented
+
+        return (
+            self.expr.isidentical(other.expr) and
```

Comment:
Are we not using `_expr_eq` here to avoid the extra cost of the two `is None` checks? That seems unlikely to be worth the cost over the improved clarity of making it clear that we want to do the logical comparison for all three blaze expressions. At the very least, we should comment here that these are the same logical check (assuming that's the case and I'm not missing something). If the perf does matter that much, should we be binding `_expr_eq` as a local?

Reply:
`expr` can't be None so I didn't consider using that. I could use the method to be more clear though.